File: component/CommonComponent/src/main/java/com/energy/commoncomponent/view/tempcanvas/BaseView.java
package com.energy.commoncomponent.view.tempcanvas;

import android.graphics.Bitmap;
import android.graphics.Matrix;

/**
 * Created by fengjibo on 2023/6/28.
 */
public abstract class BaseView {
    protected String mId;
    protected String mLabel; //标记内容
    protected String mNote; //备注
    protected double mMaxTemp;//最大温度
    protected double mMinTemp;//最大温度
    protected double mAvgTemp;//最小温度
    protected int mPointSize = 0;
    public String getId() {
        return mId;
    }

    public void setId(String id) {
        this.mId = id;
    }

    public String getLabel() {
        return mLabel;
    }

    public void setLabel(String label) {
        this.mLabel = label;
    }

    public double getMaxTemp() {
        return mMaxTemp;
    }

    public void setMaxTemp(double maxTemp) {
        this.mMaxTemp = maxTemp;
    }

    public double getMinTemp() {
        return mMinTemp;
    }

    public void setMinTemp(double minTemp) {
        this.mMinTemp = minTemp;
    }

    public double getAvgTemp() {
        return mAvgTemp;
    }

    public void setAvgTemp(double avgTemp) {
        this.mAvgTemp = avgTemp;
    }

    public String getNote() {
        return mNote;
    }

    public void setNote(String note) {
        this.mNote = note;
    }


    public Bitmap getCustomSizeImg(Bitmap rootImg, int goalW, int goalH) {
        int rootW = rootImg.getWidth();
        int rootH = rootImg.getHeight();
        Matrix matrix = new Matrix();
        matrix.postScale(goalW * 1.0f / rootW, goalH * 1.0f / rootH);
        return Bitmap.createBitmap(rootImg, 0, 0, rootW, rootH, matrix, true);
    }
}

File: android/libui/src/main/java/com/energy/view/tempcanvas/BaseView.kt
package com.energy.view.tempcanvas

import android.graphics.Bitmap
import android.graphics.Matrix

/**
 * Created by fengjibo on 2023/6/28.
 */
abstract class BaseView {
    protected var mId: String = ""
    protected var mPointSize = 0
    
    var label: String = "" // 标记内容
    var note: String = "" // 备注
    var maxTemp: Double = 0.0 // 最大温度
    var minTemp: Double = 0.0 // 最小温度
    var avgTemp: Double = 0.0 // 平均温度

    fun getId(): String = mId
    fun setId(id: String) {
        this.mId = id
    }

    protected fun getCustomSizeImg(rootImg: Bitmap, goalW: Int, goalH: Int): Bitmap {
        val rootW = rootImg.width
        val rootH = rootImg.height
        val matrix = Matrix().apply {
            postScale(goalW * 1.0f / rootW, goalH * 1.0f / rootH)
        }
        return Bitmap.createBitmap(rootImg, 0, 0, rootW, rootH, matrix, true)
    }
}